### PR TITLE
Add pull request template

### LINKED ISSUE #53: Add pull request template for better descriptions
As a reviewer in this repo, I have noticed a struggle to write clear and concise pull request descriptions. This can make it difficult for reviewers to understand the purpose and scope of the proposed changes.

To help address this issue, I recommend adding a pull request template.
A template would provide a structured format for writing descriptions, which can improve clarity and consistency.

The template could include sections for:

- Summary of changes: A brief overview of what the pull request accomplishes
- Context: The problem or issue that the changes aim to solve
- Implementation details: Any technical details or considerations that reviewers should be aware of
- Testing plan: Steps taken to ensure the changes have been properly tested and verified

Templates help you better understand what information you should include in your description and help reviewers evaluate your proposed changes.
This improves the overall quality of submissions and makes the review process more efficient and effective. 

Please let me know what you think.

### PR DESCRIPTION
**Pull request by IdanZimi**
GitHub automatically looks for certain files in the .github directory of the repository, including pull request templates. If there is a file named PULL_REQUEST_TEMPLATE.md in the .github directory, it will be used as the default pull request template for your repository. When someone opens a new pull request, they will see the contents of this file as the default description.

closes #53

### CODE REVIEW

**Review by Danielsio — APPROVED** [review]

**Comment by Danielsio** [comment]
Nice idan, this is really important, Good work !

**Review by Yuval-Vino — COMMENTED** [review]
Thanks for addressing the issue,  look good!
how about adding a section `Dependencies` If the changes have any dependencies, such as other PRs

**Comment by IdanZimi** [comment]
> Thanks for addressing the issue, look good! how about adding a section `Dependencies` If the changes have any dependencies, such as other PRs

Good idea, will add section.

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+Resolves #[relevant issue(s), e.g. 47]
+
+### Dependencies
+
+This section inidicates whether this PR depends on any other PR's. 
+Might need to wait for other PR's to merged before. Add the PR ID.
+- [ ] This PR depends on other PR's
+  changes
+- [ ] This PR is not depend on other PR's
```
Comment (yftacherzog) [inline .github/pull_request_template.md:24]:
```suggestion
- [ ] This PR does not depend on other PR's 
```

Comment (IdanZimi) [reply .github/pull_request_template.md]: Thank you. changed. @yftacherzog

**Review by SchSeba — APPROVED** [review]
/lgtm

**Review by yftacherzog — APPROVED** [review]

**Pull request merged**